### ISSUE TITLE
Improve including source files from outside of project folder using wildcards

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/sdk#551]
When people want to include files in their projects from outside the project folder (for example shared utilities code), they will likely try to do so by adding something like the following to the project file:

```xml
<Compile Include="..\..\Shared\**\*.cs" />
```

This ends up being represented in the Solution Explorer project tree with the relative path to the items, like this:

![image](https://cloud.githubusercontent.com/assets/145043/21488568/2bc25ec8-cb96-11e6-8080-718c7af05781.png)

Ideally, the relative path to the items would not be represented in the project tree.  We might be able to do this in the .NET SDK targets, by adding `Link` metadata to each item that doesn't have it and is outside of the project tree.  (This currently appears to be blocked by dotnet/roslyn-project-system#1041).

It would also be nice to be able to specify where in the project tree the linked items should be placed, while keeping their filenames and relative paths.  We could support this with a new piece of metadata:

```xml
<Compile Include="..\..\Shared\**\*.cs" BaseLink="Shared" />
```

**Comment by dsplaisted** [comment]
Closing this in favor of #1115, which covers the same scenario.
